packages/kbn-apm-synthtrace/src/lib/apm/client/entities_synthtrace_kibana_client.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import fetch from 'node-fetch';
+import { Logger } from '../../utils/create_logger';
+import { kibanaHeaders } from '../../shared/client_headers';
+import { getFetchAgent } from '../../../cli/utils/ssl';
+
+interface EntityDefinitionResponse {
+  definitions: Array<{ type: string; state: { installed: boolean; running: boolean } }>;
+}
+
+export class EntitiesSynthtraceKibanaClient {
+  private readonly logger: Logger;
+  private target: string;
+
+  constructor(options: { logger: Logger; target: string }) {
+    this.logger = options.logger;
+    this.target = options.target;
+  }
+
+  async installEntityIndexPatterns() {
+    const url = `${this.target}/internal/entities/definition?includeState=true`;
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: kibanaHeaders(),
+      agent: getFetchAgent(url),
+    });
+    const entityDefinition: EntityDefinitionResponse = await response.json();
+
+    const hasServiceEntityDefinition = entityDefinition.definitions.find(
+      (definition) => definition.type === 'service'
+    )?.state.installed;
+
+    if (hasServiceEntityDefinition === true) {
+      this.logger.debug('Service Entity is already defined');
+    } else {
+      this.logger.debug('Installing Service Entity definition');
+      const entityEnablementUrl = `${this.target}/internal/entities/managed/enablement?installOnly=true`;
+      await fetch(entityEnablementUrl, {
+        method: 'PUT',
+        headers: kibanaHeaders(),
+        agent: getFetchAgent(url),
+      });
+    }
+  }
+
+  async uninstallEntityIndexPatterns() {
```

Comment:
is it used anywhere?

Reply:
No, but tests could use it `after` they run.